lib/chef/provider/powershell_script.rb

```diff
@@ -24,71 +24,153 @@ class PowershellScript < Chef::Provider::WindowsScript
 
       provides :powershell_script, os: "windows"
 
+      def initialize (new_resource, run_context)
+        super(new_resource, run_context, '.ps1')
+        add_exit_status_wrapper
+      end
+
+      def action_run
+        valid_syntax = validate_script_syntax!
+        super if valid_syntax
```

Comment:
Good question which is related to @ksubrama's. Either we should just assume exception solves everything, or we should make this evident in the code / comments.